### PR TITLE
EDSC-3837: Display MBR and warning when a MBR will be sent to a Harmony service

### PR DESCRIPTION
**Pull request by macrouch**
# Overview

### What is the feature?

When a user is configuring a Harmony service that accepts bounding box spatial and not shapefile spatial we already convert their applied spatial into a minimum bounding rectangle (MBR). However, we don't tell the user that we are doing that, resulting in subsetted data that is not what the user expects.

This PR adds a warning message to the "Spatial Subsetting" option on the project page of a selected Harmony service. It also adds a red MBR outline to the map. The warning message is specific to the type of spatial the user has applied.

When the criteria are not met, do not display a warning message and do not show an MBR on the map.

Within this PR I also refactored how we are handling MBRs on the map. Every time we draw a spatial I am calculating the MBR and saving it to the drawnLayers object as `layerMbr`. When the various criteria are met for displaying the MBR I am adding it to the map, and removing it from the map when necessary, but I am not destroying the layer.

I also fixed an issue where the URL parameter for the Harmony spatial subsetting checkbox was not working as expected. My Playwright tests show it working as expected when loading the page with it enabled and disabled.

### What areas of the application does this impact?

Harmony access methods

# Testing

These links will load a collection with a Harmony job that supports bounding box, but not shapefile subsetting, with various spatial applied
- Polygon: http://localhost:8080/projects?p=C2930725014-LARC_CLOUD!C2930725014-LARC_CLOUD&pg[1][v]=t&pg[1][m]=harmony0&pg[1][cd]=t&pg[1][ets]=t&pg[1][ess]=t&q=TEMPO_NO2_L2&polygon[0]=-80.01563%2C38.50941%2C-82.96875%2C37.24411%2C-83.53125%2C34.7135%2C-80.15625%2C34.15114%2C-78.1875%2C33.86996%2C-79.73438%2C36.25998%2C-80.01563%2C38.50941&qt=2024-10-30T23%3A55%3A54.901Z%2C2024-10-31T20%3A05%3A11.675Z&ff=Customizable&tl=1731618651!3!!&lat=30.593395788325108&long=-135
- Point: http://localhost:8080/projects?p=C2930725014-LARC_CLOUD!C2930725014-LARC_CLOUD&q=TEMPO_NO2_L2&sp[0]=-80.22656%2C40.30775&qt=2024-10-30T23%3A55%3A54.901Z%2C2024-10-31T20%3A05%3A11.675Z&ff=Customizable&tl=1731618651!3!!&lat=40.7021484375&long=-88.787109375&zoom=5
- Circle: http://localhost:8080/projects?p=C2930725014-LARC_CLOUD!C2930725014-LARC_CLOUD&q=TEMPO_NO2_L2&circle[0]=-79.17188%2C40.22307%2C66953&qt=2024-10-30T23%3A55%3A54.901Z%2C2024-10-31T20%3A05%3A11.675Z&ff=Customizable&tl=1731618651!3!!&lat=40.01756301455347&long=-132.75 
- Line: http://localhost:8080/projects?p=C2930725014-LARC_CLOUD!C2930725014-LARC_CLOUD&q=TEMPO_NO2_L2&line[0]=-106%2C35%2C-105%2C36%2C-94%2C33%2C-95%2C30%2C-93%2C31%2C-92%2C30&qt=2024-10-30T23%3A55%3A54.901Z%2C2024-10-31T20%3A05%3A11.675Z&ff=Customizable&tl=1732138274!3!!&lat=25.376367541292524&long=-135.984375
- Bounding Box: http://localhost:8080/projects?p=C2930725014-LARC_CLOUD!C2930725014-LARC_CLOUD&pg[1][v]=t&q=TEMPO_NO2_L2&sb[0]=-79.45313%2C35.84682%2C-79.03125%2C37.11175&qt=2024-10-30T23%3A55%3A54.901Z%2C2024-10-31T20%3A05%3A11.675Z&ff=Customizable&tl=1731618651!3!!&lat=38.04833404190873&long=-146.53125
  - (does not display a warning because it is already the correct type)

Both Harmony services on this collection support shapefile subsetting, so you will not see the MBR on the map or the warning message
http://localhost:8080/projects?p=C2237824918-ORNL_CLOUD!C2237824918-ORNL_CLOUD&pg[1][v]=t&pg[1][gsk]=-start_date&pg[1][m]=harmony0&pg[1][cd]=f&pg[1][ets]=t&pg[1][ess]=t&circle[0]=-79.3125%2C39.79495%2C119632&qt=2022-12-30T17%3A47%3A56.379Z%2C2023-01-20T05%3A42%3A11.475Z&ff=Customizable&tl=1673865780!3!!&lat=39.04326016625277&long=-147.375

### Attachments

Spatial warning and MBR outline for a polygon
Screenshot 2024-11-25 at 9 59 01 PM

No warning/MBR when subsetting is disabled
Screenshot 2024-11-25 at 9 59 09 PM

# Checklist

- [x] I have added automated tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings

### CODE REVIEW

File: static/src/js/components/AccessMethod/__tests__/AccessMethod.test.js [hunk 1]
```diff
@@ -1181,187 +1187,195 @@ describe('AccessMethod component', () => {
           expect(screen.getByRole('checkbox').checked).toEqual(false)
         })
 
-        describe('when enableSpatialSubsetting is set to false', () => {
-          test('sets the checkbox unchecked for boundingBox', () => {
-            setup({
+        describe('when the user checks enableTemporalSubsetting', () => {
+          test('sets the checkbox for temporal unchecked', async () => {
+            const user = userEvent.setup()
```
Comment (macrouch) [inline static/src/js/components/AccessMethod/__tests__/AccessMethod.test.js:1192]: Don't need this

File: static/src/js/components/SpatialSelection/SpatialSelection.jsx [hunk 2]
```diff
@@ -788,6 +794,36 @@ const SpatialSelection = (props) => {
...
+      const correctLayer = points === layerPoints
+
+      // If we have the correct layer and the layerMbr isn't already on the map, add it
+      if (correctLayer && layerMbr && !layerMbr._map) {
+        layerMbr.addTo(featureGroupRef.current)
+      }
+    })
+  }
+
+  const hideLayerMbr = (points) => {
```
Comment (macrouch) [inline static/src/js/components/SpatialSelection/SpatialSelection.jsx:812]: Add comment

**Review by eudoroolivares2016 — APPROVED** [review]

**Comment by eudoroolivares2016** [comment]
Tested locally all the links and some others on the PR Url parameters for spatial and temporal do work as expected this is much cleaner now tried with multi shape shapefiles as well

**Review by dmistry1 — APPROVED** [review]
there are couple `setupTests` in tour.spec.js with two `awaits`

**Pull request merged**